### PR TITLE
Trashcoder/blob slice content type

### PR DESCRIPTION
**Pull request by trashcoder**
## Summary:
I added the contentType parameter to Blob.slice like it's in the MDN Web docs. 
This PR fixes https://github.com/facebook/react-native/issues/38058

When i slice a Blob for chunked uploads with react native i lost the content type, e.g. "image/jpeg", so the server doesn't know what kind of file he gets. In the docs of MDN the slice method was described with a third contentType parameter which was missing in Metas implementation.
## Changelog:
 [GENERAL] [ADDED] added a third parameter "contentType" to method slice of class Blob.

## Test Plan:
I tested it with the unit-tests:
yarn run test Blob-test.js
yarn run v1.22.19
$ jest Blob-test.js
 PASS  packages/react-native/Libraries/Blob/__tests__/Blob-test.js
  Blob
    ✓ should create empty blob (5 ms)
    ✓ should create blob from other blobs and strings
    ✓ should slice a blob (1 ms)
    ✓ should slice a blob and sets a contentType   
    ✓ should close a blob (4 ms)

My added unit test results "✓ should slice a blob and sets a contentType".

### CODE REVIEW

**Comment by trashcoder** [comment]
Is there something i can do to make the PR pass? It seems these 3 fails were there because of a server error.

**Comment by Pranav-yadav** [comment]
Congrats on your 1st PR to RN 🎉

> Is there something i can do to make the PR pass? It seems these 3 fails were there because of a server error.

Those seem transient issues, unrelated to changes in this PR so no need to worry about them.
Apart from that could you please link the respective issue to this PR?  By mentioning below in the PR description:
Fixes #38058

File: packages/react-native/Libraries/Blob/__tests__/Blob-test.js [hunk 1]
```diff
@@ -77,6 +77,17 @@ describe('Blob', function () {
     expect(sliceC.data.offset).toBe(34543);
     expect(sliceC.size).toBe(Math.min(blob.data.size, 34569) - 34543);
   });
+  it('should slice a blob and sets a contentType', () => {
```
Comment (Pranav-yadav) [inline packages/react-native/Libraries/Blob/__tests__/Blob-test.js:80]:
```suggestion
  });
  
  it('should slice a blob and sets a contentType', () => {
```

**Review by Pranav-yadav — CHANGES_REQUESTED** [review]
Left format suggestions and tips.

Also, something important this PR doesn't add is "negative tests".
We should always add the tests for negative cases e.g. _malformed_ & _empty_ `contentType`

P.S.: This practice has helped me avoid potential regressions 😅

Overall, this is awesome ❤️

File: packages/react-native/Libraries/Blob/__tests__/Blob-test.js [hunk 2]
```diff
@@ -77,6 +77,18 @@ describe('Blob', function () {
     expect(sliceC.data.offset).toBe(34543);
     expect(sliceC.size).toBe(Math.min(blob.data.size, 34569) - 34543);
   });
+
```
[the remark this answers is not in the document]
Comment (Pranav-yadav) [reply packages/react-native/Libraries/Blob/__tests__/Blob-test.js:80]: @trashcoder You should always run `yarn lint && yarn prettier` before commiting 🚀 :)

[the remark this answers is not in the document]
Comment (trashcoder) [reply packages/react-native/Libraries/Blob/__tests__/Blob-test.js:80]: Can i still make these changes with yarn lint and yarn prettier? or would that be another PR?

**Comment by Pranav-yadav** [comment]
>> @trashcoder You should always run `yarn lint && yarn prettier` before commiting 🚀 :)
>
> Can i still make these changes with yarn lint and yarn prettier? or would that be another PR?

In this PR only. It's part of the development Workflow you can check the the guide here: https://reactnative.dev/contributing/how-to-contribute-code#development-workflow

For the scope of this PR below commands should suffice:
- running `yarn lint` will run `eslint` for JS, TS, etc. which performs linting/static analysis and makes sure our code bug free. 😃
- running `yarn prettier` will run `prettier` which is used to format/prettify our code 💅
- running `yarn test` will run the existing using `jest`; the existing tests and new ones added by you.

Let me know if you need any further help.

File: packages/react-native/Libraries/Blob/Blob.js [hunk 3]
```diff
@@ -81,7 +81,7 @@ class Blob {
     return this._data;
   }
 
-  slice(start?: number, end?: number): Blob {
+  slice(start?: number, end?: number, contentType: string = ''): Blob {
     const BlobManager = require('./BlobManager');
     let {offset, size} = this.data;
```
Comment (javache) [inline packages/react-native/Libraries/Blob/Blob.js:86]: Should we inherit the previous type by default?
```suggestion
    let {offset, size, type} = this.data;
```

Comment (blakef) [reply packages/react-native/Libraries/Blob/Blob.js:86]: @javache I'm not sure we should, the spec says to initialise this as [`null`](https://w3c.github.io/FileAPI/#slice-method-algo).  We differ from the semantics of the File Blob definition, but this is for performance reasons right?

**Comment by blakef** [comment]
LGTM, fix the formatting and we should get this in.

**Comment by trashcoder** [comment]
@blakef formatting is already fixed.
